src/profile-logic/import/chrome.js

```diff
@@ -390,172 +419,181 @@ async function processTracingEvents(
+          frameTable.optimizations[frameIndex] = null;
+          frameTable.length = Math.max(frameTable.length, frameIndex + 1);
+
+          stackTable.frame.push(frameIndex);
+          stackTable.category.push(category);
+          stackTable.subcategory.push(0);
+          stackTable.prefix.push(prefixStackIndex);
+          nodeIdToStackId.set(nodeIndex, stackTable.length++);
         }
-        frameTable.address[frameIndex] = -1;
-        frameTable.category[frameIndex] = category;
-        frameTable.subcategory[frameIndex] = 0;
-        frameTable.func[frameIndex] = funcId;
-        frameTable.innerWindowID[frameIndex] = 0;
-        frameTable.implementation[frameIndex] = null;
-        frameTable.line[frameIndex] =
-          lineNumber === undefined ? null : lineNumber;
-        frameTable.column[frameIndex] =
-          columnNumber === undefined ? null : columnNumber;
-        frameTable.optimizations[frameIndex] = null;
-        frameTable.length = Math.max(frameTable.length, frameIndex + 1);
-
-        stackTable.frame.push(frameIndex);
-        stackTable.category.push(category);
-        stackTable.subcategory.push(0);
-        stackTable.prefix.push(prefixStackIndex);
-        nodeIdToStackId.set(nodeIndex, stackTable.length++);
       }
-    }
 
-    // Chrome profiles sample much more frequently than Gecko ones do, and they store
-    // the time delta between each sampling event. In order to properly reconstruct
-    // the data using our fixed-time intervals, sample the data at a fixed rate that
-    // is most likely slightly higher. Chrome profiles have been observed sampling
-    // between 100us to 300us. Reconstruct the profile at 500us, which is a somewhat
-    // reasonable interval.
-
-    // Choose 500us as a somewhat reasonable sampling interval. When converting
-    // the chrome profile, this function samples the chrome profile, and generates
-    // new samples on our target interval of 500us.
-    profile.meta.interval = 0.5;
-
-    for (let i = 0; i < samples.length; i++) {
-      const nodeIndex = samples[i];
-      // Convert to milliseconds:
-      threadInfo.lastSeenTime += timeDeltas[i] / 1000;
-      if (
-        threadInfo.lastSeenTime - threadInfo.lastSampledTime >=
-        profile.meta.interval
-      ) {
-        threadInfo.lastSampledTime = threadInfo.lastSeenTime;
-        const stackIndex = ensureExists(
-          nodeIdToStackId.get(nodeIndex),
-          'Could not find the stack information for a sample when decoding a Chrome profile.'
-        );
-        ensureExists(
-          samplesTable.eventDelay,
-          'Could not find the eventDelay in samplesTable inside the newly created Chrome profile thread.'
-        ).push(null);
-        samplesTable.stack.push(stackIndex);
-        samplesTable.time.push(threadInfo.lastSampledTime);
-        samplesTable.length++;
+      // Chrome profiles sample much more frequently than Gecko ones do, and they store
```

Comment:
No action needed: Is there any way we can get closer to just using their data, rather than fake-sampling it?

Reply:
I think the Chrome profiler is a sampling-based profiler these days, so we might be able to turn our fake-sampling into real-sampling; i.e. follow the actual sample timestamps. The Android trace importer PR has an implementation of that.
If the Chrome profiler is a tracing-based profiler, I'm not sure how we should represent the data. We could use the duration field and have tons of short samples; it's worth trying out how well that works.